File: server/order-service/src/main/java/com/sudo248/orderservice/controller/order/dto/CartProductDto.java
package com.sudo248.orderservice.controller.order.dto;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class CartProductDto {
    private String cartProductId;
    private String cartId;
    private String productId;
    protected int quantity;
}

File: server/cart-service/src/main/kotlin/com/sudoo/cartservice/controller/dto/CartProductDto.kt
package com.sudoo.cartservice.controller.dto

import com.sudoo.cartservice.repository.entity.CartProduct

data class CartProductDto(
    val cartProductId: String = "",
    val cartId: String = "",
    val quantity: Int = 0,
    val totalPrice: Double = 0.0,
    val purchasePrice:Double? = null,
    val product: ProductInfoDto? = null
)

fun CartProductDto.toCartProduct(): CartProduct {
    return CartProduct(
        cartProductId = this.cartProductId,
        cartId = this.cartId,
        purchasePrice = this.purchasePrice,
        quantity = this.quantity,
        productId = product?.productId ?: ""
    )
}